### PR TITLE
disable sloppy mode block level functions when not bundling

### LINKED ISSUE #2862: Cannot declare a var variable that shadows a let/const/class variable: 'cb'. while parsing module
### What version of Bun is running?

canary 0.6.0

### What platform is your computer?

linux deb

### What steps can reproduce the bug?

this error came as a result of trying bookshelf orm on bun.

steps to reproduce

bun install knex bookshelf elysia

import knex from "knex";
import bookshelf from "bookshelf";

console.log(bookshelf, knex); // functions

knex({
  client: "sqlite3",
  connection: {
    host: "127.0.0.1",
    charset: "utf8",
  },
});

### What is the expected behavior?

Knex is a purpolar sql query builder for js, we need it to work in bun.

knex should succeccfully initialise the db amd be able to run transactions on bun.

### What do you see instead?

// error

```
[Function] [Function]
Knex: run
$ npm install sqlite3 --save
Cannot declare a var variable that shadows a let/const/class variable: 'cb'. while parsing module "/project/node_modules/sqlite3/lib/sqlite3.js"
loadCJS2ESM@[native code]
requireESM@[native code]
internalRequire@[native code]
require@[native code]
initializeDriver@/project/node_modules/knex/lib/client.js:136:35
Client@/project/node_modules/knex/lib/client.js:53:34
Client_SQLite3@/project/node_modules/knex/lib/dialects/sqlite3/index.js:22:12
knex@/project/node_modules/knex/lib/knex-builder/Knex.js:6:100
module code@/project/DB/models.js:173:5
evaluate@[native code]
moduleEvaluation@[native code]
moduleEvaluation@[native code]
@[native code]
asyncFunctionResume@[native code]
asyncModuleEvaluation@[native code]
@[native code]
189 |     try {
190 |       this.driver = this._driver();
191 |     } catch (e) {
192 |       const message = `Knex: run\n$ npm install ${this.driverName} --save`;
193 |       this.logger.error(`${message}\n${e.message}\n${e.stack}`);
194 |       throw new Error(`${message}\n${e.message}`);
                ^
error: Knex: run
$ npm install sqlite3 --save
Cannot declare a var variable that shadows a let/const/class variable: 'cb'. while parsing module "/project/node_modules/sqlite3/lib/sqlite3.js"
      at initializeDriver (/project/node_modules/knex/lib/client.js:194:12)
      at new Client (/project/node_modules/knex/lib/client.js:75:6)
      at new Client_SQLite3 (/project/node_modules/knex/lib/dialects/sqlite3/index.js:22:4)
      at /project/node_modules/knex/lib/knex-builder/Knex.js:16:27
      at /project/DB/models.js:9:0
189 |     try {
190 |       this.driver = this._driver();
191 |     } catch (e) {
192 |       const message = `Knex: run\n$ npm install ${this.driverName} --save`;
193 |       this.logger.error(`${message}\n${e.message}\n${e.stack}`);
194 |       throw new Error(`${message}\n${e.message}`);
                ^
error: Knex: run
$ npm install sqlite3 --save
Cannot declare a var variable that shadows a let/const/class variable: 'cb'. while parsing module "/project/node_modules/sqlite3/lib/sqlite3.js"
      at initializeDriver (/project/node_modules/knex/lib/client.js:194:12)
      at new Client (/project/node_modules/knex/lib/client.js:75:6)
      at new Client_SQLite3 (/project/node_modules/knex/lib/dialects/sqlite3/index.js:22:4)
      at /project/node_modules/knex/lib/knex-builder/Knex.js:16:27
      at /project/DB/models.js:9:0



```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dylan-conway**
fixes #2862

### CODE REVIEW

File: src/js_parser.zig [hunk 1]
```diff
@@ -6643,7 +6643,8 @@ fn NewParser_(
                         var is_sloppy_mode_block_level_fn_stmt = false;
                         const original_member_ref = value.ref;
 
-                        if (symbol.kind == .hoisted_function) {
+                        // TODO: enable this when not bundling. the name for the new symbol needs to go through a renamer
+                        if (p.options.bundle and symbol.kind == .hoisted_function) {
```
Comment (Jarred-Sumner) [inline src/js_parser.zig]: we run it through the renamer when `--minify-syntax`

maybe add a `p.willRenameSymbols()`?

Comment (dylan-conway) [reply src/js_parser.zig]: added `willUseNumberRenamer()`. also I should've been more specific, it needs to go through the number renamer.

File: src/js_parser.zig [hunk 2]
```diff
@@ -6593,6 +6593,10 @@ fn NewParser_(
             // p.resolveGeneratedSymbol(&p.jsx_filename);
         }
 
+        fn willUseNumberRenamer(p: *P) bool {
+            return p.options.bundle;
```
Comment (Jarred-Sumner) [inline src/js_parser.zig:6597]: it should check:
- is it bundling
-if it's not bundling, is features.minify_syntax enabled

Comment (Jarred-Sumner) [inline src/js_parser.zig:6597]:
```suggestion
            return p.options.bundle or !p.options.features.minify_identifiers;
```

**Pull request merged**